lib/sisimai/bite/email/apachejames.rb

```diff
@@ -0,0 +1,169 @@
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+            next if e.empty?
+
+            # Message details:
+            #   Subject: Nyaaan
+            #   Sent date: Thu Apr 29 01:20:50 JST 2015
+            #   MAIL FROM: shironeko@example.jp
+            #   RCPT TO: kijitora@example.org
+            #   From: Neko <shironeko@example.jp>
+            #   To: kijitora@example.org
+            #   Size (in bytes): 1024
+            #   Number of lines: 64
+            v = dscontents[-1]
+
+            if cv = e.match(/\A[ ][ ]RCPT[ ]TO:[ ]([^ ]+[@][^ ]+)\z/)
+              #   RCPT TO: kijitora@example.org
+              if v['recipient']
+                # There are multiple recipient addresses in the message body.
+                dscontents << Sisimai::Bite.DELIVERYSTATUS
+                v = dscontents[-1]
+              end
+              v['recipient'] = cv[1]
+              recipients += 1
+
+            elsif cv = e.match(/\A[ ][ ]Sent[ ]date:[ ](.+)\z/)
+              #   Sent date: Thu Apr 29 01:20:50 JST 2015
+              v['date'] = cv[1]
+
+            elsif cv = e.match(/\A[ ][ ]Subject:[ ](.+)\z/)
+              #   Subject: Nyaaan
+              subjecttxt = cv[1]
+
+            else
+              next if gotmessage == 1
+              if v['diagnosis']
+                # Get an error message text
+                if e =~ /\AMessage[ ]details:\z/
+                  # Message details:
+                  #   Subject: nyaan
+                  #   ...
+                  gotmessage = 1
+                else
+                  # Append error message text like the followng:
+                  #   Error message below:
+                  #   550 - Requested action not taken: no such user here
+                  v['diagnosis'] ||= ''
+                  v['diagnosis']  += ' ' + e
+                end
+
+              else
+                # Error message below:
+                # 550 - Requested action not taken: no such user here
+                v['diagnosis'] = e if e =~ Re1[:error]
+              end
+            end
+          end
+        end
+        return nil if recipients.zero?
+        require 'sisimai/string'
+
+        unless rfc822list.find { |a| a =~ /^Subject:/ }
+          # Set the value of $subjecttxt as a Subject if there is no original
+          # message in the bounce mail.
+          rfc822list << sprintf('Subject: %s', subjecttxt)
```

Comment:
__[rubocop]__
* Style/IfInsideElse: Convert `if` nested inside `else` to `elsif`.

<!-- Comment by [SideCI](https://sideci.com) -->